terraform/modules/storage/s3-private/variables.tf

```diff
@@ -0,0 +1,24 @@
+variable "environment" {
+  description = "The Deployment environment"
+  type = string
+}
+
+variable "region" {
+  description = "The AWS region"
+  type = string
+}
+
+variable "domain" {
+  description = "The domain of the S3 bucket"
+  type = string
+}
+
+variable "service_name" {
+  description = "The name of the service for which the bucket supplies"
+  type = string
+}
+
+variable "lifecycled_prefixes" {
+  description = "The data-slash prefixes requiring lifecycles"
+  type = list(string)
```

Comment:
can you have the option of none? If we want to keep some buckets all the time

Reply:
Yes, an empty list does that [], like in the case for the s3-ml one in main

Reply:
Also, I think this only setting lifecycles on prefixes within the bucket, as opposed to on the entire bucket